presentation/src/main/java/com/sopt/presentation/appointmentCreate/appointmentCreatePeriod/AppointmentCreatePeriodRoute.kt

```diff
@@ -0,0 +1,194 @@
+import com.sopt.core.designsystem.component.calendar.NoostakCalendar
+import com.sopt.core.designsystem.component.progressbar.NoostakProgressBar
+import com.sopt.core.designsystem.component.text.NoostakHeaderText
+import com.sopt.core.designsystem.component.toggle.NoostakSwitch
+import com.sopt.core.designsystem.component.topappbar.NoostakTopAppBar
+import com.sopt.core.designsystem.theme.NoostakTheme
+import com.sopt.presentation.R
+
+@Composable
+fun AppointmentCreatePeriodRoute(
+    groupId: Long,
+    appointmentName: String,
+    appointmentCategory: String,
+    appointmentTime: Int,
+    navigateUp: () -> Unit,
+    navigateToTimePicker: (Long, String, String, Int, Boolean, List<String>) -> Unit,
+    calendarPeriodViewModel: AppointmentCreatePeriodViewModel = hiltViewModel()
+) {
+    LaunchedEffect(key1 = calendarPeriodViewModel.sideEffects) {
+        calendarPeriodViewModel.sideEffects.collect { sideEffect ->
+            when (sideEffect) {
+                is AppointmentCreatePeriodSideEffect.NavigateToTimePicker -> {
+                    navigateToTimePicker(
+                        sideEffect.groupId,
+                        sideEffect.appointmentName,
+                        sideEffect.appointmentCategory,
+                        sideEffect.appointmentDuration,
+                        sideEffect.isSingleDateMode,
+                        sideEffect.appointmentDate
+                    )
+                }
+                is AppointmentCreatePeriodSideEffect.NavigateUp -> {
+                    navigateUp()
+                }
+            }
+        }
+    }
+
+    AppointmentCreatePeriodScreen(
+        onBackButtonClick = calendarPeriodViewModel::navigateUp,
+        onButtonClick = calendarPeriodViewModel::navigateToAppointmentCreateTimePicker,
+        appointmentName = appointmentName,
+        appointmentCategory = appointmentCategory,
+        appointmentDuration = appointmentTime,
+        appointmentDate = calendarPeriodViewModel.days,
+        groupId = groupId
+    )
+}
+
+@Composable
+fun AppointmentCreatePeriodScreen(
+    onButtonClick: (Long, String, String, Int, Boolean, List<String>) -> Unit,
+    onBackButtonClick: () -> Unit,
+    appointmentName: String,
+    appointmentCategory: String,
+    appointmentDuration: Int,
+    appointmentDate: List<String>,
+    groupId: Long
+) {
+    var startDate by remember { mutableStateOf("") }
+    var endDate by remember { mutableStateOf("") }
+    var dates by remember { mutableStateOf(listOf<String>()) }
+    var isSingleDateMode by remember { mutableStateOf(false) }
+    var isButtonEnabled by remember { mutableStateOf(false) }
+
+    val typography = NoostakTheme.typography
+    val colors = NoostakTheme.colors
+
+    LaunchedEffect(isSingleDateMode, dates) {
+        isButtonEnabled = if (isSingleDateMode) {
+            dates.size == 1
+        } else {
+            dates.isNotEmpty()
+        }
+    }
```

Comment:
P1: 1일만 선택해야 버튼 활성화되는 오류 해결해주세요. in 1..7로 변경하시면 됩니다
```suggestion
    LaunchedEffect(isSingleDateMode, dates) {
        isButtonEnabled = if (isSingleDateMode) {
            dates.size in 1..7
        } else {
            dates.isNotEmpty()
        }
    }
```

Reply:
수정완!